presto-druid/src/main/java/com/facebook/presto/druid/DruidClient.java

```diff
@@ -119,21 +124,26 @@ public DruidSegmentInfo getSingleSegmentInfo(String dataSource, String segmentId
         return httpClient.execute(request, createJsonResponseHandler(SEGMENT_INFO_CODEC));
     }
 
-    public String getData(String dql)
+    public InputStream getData(String dql)
     {
-        return httpClient.execute(prepareQuery(dql), createStringResponseHandler()).getBody();
```

Comment:
I would suggest, rename prepareQuery to prepareMetadataQuery
and rename prepareQueryWithLines to prepareDataQuery
what do you think?